### ISSUE TITLE
Add TSMP documentation as GitHub Page

### ISSUE THREAD

**Issue by niklaswr** [HPSCTerrSys/TSMP#159]
## Add TSMP documentation as [GitHub Page](https://pages.github.com/)

There is a general and long-standing need for a central location for TSMP documentation.  
As an example of how old this need is, there is a two year old [milestone](https://github.com/HPSCTerrSys/TSMP/milestone/1) that also addresses this issue, which in turn is based on an even older Bugzilla entry.

Since the very beginning of TSMP several documentation approaches were started leading to several different and wide spread TSMP documentation files. The most recent and most promising attempt to gather these files into a single documentation is given by the [TSMP_manual](https://gitlab.jsc.fz-juelich.de/sdlts/tsmp_manual) repo, which is based on [markdown files](https://markdown.de/) and the documentation generator [sphinx](https://www.sphinx-doc.org/en/master/usage/quickstart.html).   
Further [GitLab CI/CD](https://docs.gitlab.com/ee/ci/) functionality is used, to automatically render the raw markdown files with sphinx and publish via [this separate GitLab page](https://sdlts.pages.jsc.fz-juelich.de/tsmp_manual/), accessible from the entire www, showing in the well known Read-The-Docs theme.   

**This approach is very interesting and I propose to adapt and port this to GitHub, shipping and maintaining the TSMP documentation with the same repository as the source code itself.**

Further I do suggest to not finalize the current [TSMP_manual](https://gitlab.jsc.fz-juelich.de/sdlts/tsmp_manual) before bringing this to GitHub, but to implement an empty GitHub Page based documentation, filling on demand.   
TSMP is developing and therefore the documentation is a living document as well. We simply dont know how the documentation will look like and therefore we cannot finalize this at a separate place before bringing it to GitHub. Further developing new documentation parts outside of GitHub and porting them back later in a separate step, is a double effort. Just start the documentation publicly, agree mistakes and uncompleted documentation, but put a value on the effort immediately. 

I will provide a pull request showing how to implement GitHub Pages, as well as a strategy of how to maintain this kind of GitHub Page based documentation.   
In the meantime I do ask you to leave a comment discussing this issue.

**Comment by niklaswr** [comment]
Pull request showing how to implement above mentioned GitHub Page based TSMP documentation is created.  
This pull request comes with a small example of documentation, which can be seen on the [GitHub page of my fork](https://niklaswr.github.io/TSMP/content/contributing/documentation.html#).

**Comment by DCaviedesV** [comment]
I agree in full with this idea. The guidelines on how to contribute are also very nice.
Any reason we cannot already migrate the [TSMP manual](https://gitlab.jsc.fz-juelich.de/sdlts/tsmp_manual) into this pull request, or after the pull request @jjokella @niklaswr ?

**Comment by jjokella** [comment]
I also agree with this initiative, having the documentation in the same repo as the code of TSMP is useful.

I fear that the way of contributing to the documentation via fork/PR might be to much effort for a general user. In the current version of the documentation I have tried to make changing the manual possible via link at the top of the page. Such a change should then not immediately be rendered, but automatically generate a PR. Could be a plan for the future to implement this also here on Github.
 
From my side, nothing speaks against porting the manual to GitHub.  Maybe as input from my side: Problems I have encountered before while porting:
- broken links due to different link syntax
- style changes (comment blocks, LaTeX etc)
- loading the correct Python modules for (version of) sphinx, see `requirements.txt` of current Gitlab

If the current TSMP-manual gets ported to Github I will move my work on TSMP-documentation there! Thanks for the effort!

**Comment by kgoergen** [comment]
I strongly agree. The full TSMP user guide should go there. Ingesting all there is so far. The FallSchool standard examples could be sections in a getting started appendix. The current README.md file would also be ingested fully. I also opt to include documentation on the complete pre-processing of the static or constant fields (e.g., soil hydraulic properties). It would also be good to include a note on the model system maintenance and configuration management.

**Comment by niklaswr** [comment]
@DCaviedesV 
From my point of view, there is no reason not to migrate the current TSMP manual to GitHub. However, from my point of view, I would not migrate the existing manual with the init pull request that implements the GitHub page-based documentation. I would start with empty documentation (except for the small "how to contribute documentation" part) and migrate the existing stuff directly via pull request. That way we have a platform to discuss potential changes (different section structure / names), we can test for the problems @jjokella is mentioning above, and we can test the strategy of doing it via pull requests right away. 

@jjokella 
I share your doubts that a PR could be a lot of work, but at the same time I think this is only a problem for the very first time. Once an 80% complete documentation is ready, there will not be too many changes, I guess. Also, if other people than the current ones start contributing, we need some workflow to manage the new income. So I think PR is the best option. However, we should test this and rethink the approach if necessary.

**Comment by s-poll** [comment]
I fully agree with the initiative and would strongly opt to include the complete current TSMP documentation in (not only the readme). Thus, we would have a one-stop-shop for all our information. Consolidation of the documentation can be done while we have the document already in the GitHub pages. 

@jjokella @niklaswr I would go for the way of PR. I now that it is some extra work, but agree that it is easier for people outside the TSMP maintainers. In my expectations the PR caused by the documentation will have its peak at the beginning of the porting

**Comment by jjokella** [comment]
I agree with @s-poll that it would be nice to have the full manual available after porting, such that I can work on the JSC-Gitlab up to a certain time and then just switch websites, but with the same content.

@niklaswr PRs are also fine from my side and I guess the main contributors are the ones also contributing to this PR anyway ;-)

**Comment by niklaswr** [comment]
All right. As far as I read comments above there is no doubt about this PR.
Since JSC is offline today, I guess this is a good opportunity to merge this PR.
However, if there is still any doubt, please stop me from merging with a comment. I plan to merge it after lunch.

**Comment by s-poll** [comment]
I am fine with merging. We should avoid to have diverging documentaries for TSMP by all means. Is everyone fine to port the current state of the TSMP documentation from JSC gitlab to here?

**Comment by niklaswr** [comment]
@s-poll fine with this, no concerns at all.  
The only restriction from me: Port the existing documentation with a separate PR, to keep this PR what it is - a technical implementation of the GitHub Pages approach.  
As I do have some capacities left today, I can create this second PR right after merging this PR.
